src/vmm/src/vcpu/mod.rs

```diff
@@ -52,39 +52,54 @@ pub enum Error {
 /// Dedicated Result type.
 pub type Result<T> = result::Result<T, Error>;
 
+pub(crate) struct VcpuState {
```

Comment:
Not sure about calling it `*State`, as state conceptually means what a vCPU is doing at a point in time, like what's in its registers etc. `VcpuConfig` is taken for the user-facing part of the VMM but this is still more of a initialization/configuration ephemeral object.

Reply:
The reason for calling it `State` is that I was looking into how we are going to implement features such as pause & resume. The general pattern that I was thinking about applying here is that the `Vcpu` and `Vm` should have only one method of initialization (i.e. new). Then, the only thing that differs from when you are starting a fresh Vcpu versus one that was resume is the `State`.

Right now since we didn't really look at the state that needs to be saved on `pause`, the vcpu & vm state is really basic, but as we add this implementation I am expecting the `State` structure to evolve such that:
- you can easily create a `default`/`empty` state; this is to be used when starting a new vm/vcpu
- add other fields to the structure related to other registers/other things that need to be saved; this should be used when resuming from a known state.

Does that make sense?